llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -12996,13 +12996,31 @@ SDValue DAGCombiner::foldPartialReduceMLAMulOp(SDNode *N) {
   SDValue Op1 = N->getOperand(1);
   SDValue Op2 = N->getOperand(2);
 
-  APInt C;
-  if (Op1->getOpcode() != ISD::MUL ||
-      !ISD::isConstantSplatVector(Op2.getNode(), C) || !C.isOne())
```

Comment:
Why do we no longer need to allow constant splats in place of the mul or shl?

Reply:
IIUC, that just moved down the file a little.